### PR TITLE
refactor: refactor test_find_date_in_string

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]

File: test/unittests/commands/excel2xml/test_excel2xml_lib.py [hunk 1]
```diff
@@ -190,126 +190,121 @@ def test_make_xsd_id_compatible(self) -> None:
...
     def test_find_date_in_string_eur_date_range(self) -> None:
         """template: 27.-28.1.1900"""
-        self.assertEqual(excel2xml.find_date_in_string("x 25.-26.2.0800 x"), "GREGORIAN:CE:0800-02-25:CE:0800-02-26")
-        self.assertEqual(excel2xml.find_date_in_string("x 25. - 26.2.0800 x"), "GREGORIAN:CE:0800-02-25:CE:0800-02-26")
-        self.assertEqual(excel2xml.find_date_in_string("x 25.-24.2.0800 x"), None)
-        self.assertEqual(excel2xml.find_date_in_string("x 25.-25.2.0800 x"), None)
+        assert excel2xml.find_date_in_string("x 25.-26.2.0800 x") == "GREGORIAN:CE:0800-02-25:CE:0800-02-26"
+        assert excel2xml.find_date_in_string("x 25. - 26.2.0800 x") == "GREGORIAN:CE:0800-02-25:CE:0800-02-26"
+        assert excel2xml.find_date_in_string("x 25.-24.2.0800 x") is None
+        assert excel2xml.find_date_in_string("x 25.-25.2.0800 x") is None
```
Comment (Nora-Olivia-Ammann) [inline test/unittests/commands/excel2xml/test_excel2xml_lib.py:211]: I don't see why this is not a valid entry. It is a time interval of one day, which is something we also do.

File: test/unittests/commands/excel2xml/test_excel2xml_lib.py [hunk 2]
```diff
@@ -190,126 +190,121 @@ def test_make_xsd_id_compatible(self) -> None:
...
-        self.assertEqual(excel2xml.find_date_in_string("x (01.03. - 25.04.2022) x"), expected)
+        assert excel2xml.find_date_in_string("x _1.3. - 25.4.2022_ x") == "GREGORIAN:CE:2022-03-01:CE:2022-04-25"
+        assert excel2xml.find_date_in_string("x (01.03. - 25.04.2022) x") == "GREGORIAN:CE:2022-03-01:CE:2022-04-25"
 
-        self.assertEqual(excel2xml.find_date_in_string("x 28.2.-1.12.1515 x"), "GREGORIAN:CE:1515-02-28:CE:1515-12-01")
-        self.assertEqual(excel2xml.find_date_in_string("x 28.2.-26.2.1515 x"), None)
-        self.assertEqual(excel2xml.find_date_in_string("x 28.2.-28.2.1515 x"), None)
+        assert excel2xml.find_date_in_string("x 28.2.-1.12.1515 x") == "GREGORIAN:CE:1515-02-28:CE:1515-12-01"
+        assert excel2xml.find_date_in_string("x 28.2.-26.2.1515 x") is None
+        assert excel2xml.find_date_in_string("x 28.2.-28.2.1515 x") is None
```
Comment (Nora-Olivia-Ammann) [inline test/unittests/commands/excel2xml/test_excel2xml_lib.py:220]: Same as above

File: test/unittests/commands/excel2xml/test_excel2xml_lib.py [hunk 3]
```diff
@@ -190,126 +190,121 @@ def test_make_xsd_id_compatible(self) -> None:
...
-        self.assertEqual(excel2xml.find_date_in_string("x 25.12.2022-03.01.2022 x"), None)
-        self.assertEqual(excel2xml.find_date_in_string("x 25.12.2022-25.12.2022 x"), None)
-        self.assertEqual(excel2xml.find_date_in_string("x 25/12/2022-03/01/2022 x"), None)
-        self.assertEqual(excel2xml.find_date_in_string("x 25/12/2022-25/12/2022 x"), None)
+        assert excel2xml.find_date_in_string("x 1.9.2022-3.1.2024 x") == "GREGORIAN:CE:2022-09-01:CE:2024-01-03"
+        assert excel2xml.find_date_in_string("x 25.12.2022 - 3.1.2024 x") == "GREGORIAN:CE:2022-12-25:CE:2024-01-03"
+        assert excel2xml.find_date_in_string("x 25/12/2022-03/01/2024 x") == "GREGORIAN:CE:2022-12-25:CE:2024-01-03"
+        assert excel2xml.find_date_in_string("x 25/12/2022 - 3/1/2024 x") == "GREGORIAN:CE:2022-12-25:CE:2024-01-03"
+        assert excel2xml.find_date_in_string("x 25.12.2022-03.01.2022 x") is None
+        assert excel2xml.find_date_in_string("x 25.12.2022-25.12.2022 x") is None
```
Comment (Nora-Olivia-Ammann) [inline test/unittests/commands/excel2xml/test_excel2xml_lib.py:229]: Same as above.

**Review by Nora-Olivia-Ammann — APPROVED** [review]

File: test/unittests/commands/excel2xml/test_excel2xml_lib.py (hunk 2 above)
Comment (jnussbaum) [reply test/unittests/commands/excel2xml/test_excel2xml_lib.py:220]: I will do that as follow-up: https://linear.app/dasch/issue/DEV-3154/featexcel2xml-find-date-in-string-allow-time-spans-of-1-day

**Pull request merged**